cmake/BLTThirdPartyConfigFlags.cmake.in

```diff
@@ -1,18 +1,29 @@
 if ("@ENABLE_CUDA@")
-    set(BLT_ENABLE_CUDA     "@ENABLE_CUDA@")
+    set(BLT_ENABLE_CUDA          "@ENABLE_CUDA@")
 endif()
 
 if ("@ENABLE_HIP@")
-    set(BLT_ENABLE_HIP      "@ENABLE_HIP@")
+    set(BLT_ENABLE_HIP           "@ENABLE_HIP@")
 endif()
 
 if ("@ENABLE_MPI@")
-    set(BLT_ENABLE_MPI      "@ENABLE_MPI@")
+    set(BLT_ENABLE_MPI           "@ENABLE_MPI@")
 endif()
 
 if ("@ENABLE_OPENMP@")
-    set(BLT_ENABLE_OPENMP   "@ENABLE_OPENMP@")
+    set(BLT_ENABLE_OPENMP        "@ENABLE_OPENMP@")
 endif()
 
-set(BLT_ENABLE_FIND_MPI     "@ENABLE_FIND_MPI@")
-set(BLT_ENABLE_CLANG_CUDA   "@ENABLE_CLANG_CUDA")
+set(BLT_ENABLE_FIND_MPI         "@ENABLE_FIND_MPI@")
+
+# User configurable CUDA options needed by downstream
+if ("@ENABLE_CLANG_CUDA")
+    set(BLT_ENABLE_CLANG_CUDA       "@ENABLE_CLANG_CUDA")
+    set(BLT_CLANG_CUDA_ARCH         "@BLT_CLANG_CUDA_ARCH@")
+endif()
+set(CUDA_TOOLKIT_ROOT_DIR       "@CUDA_TOOLKIT_ROOT_DIR@")
```

Comment:
Can you move the CUDA and HIP variables inside of the respective ENABLE sections?

Also if we are passing down the HIP arch flags we should do that for CUDA as well.  Also we should check if the downstream user set that variable and maybe not override it with ours. Though they are probably broken at that point.

Reply:
AFAIK, most projects are using newer CMake versions that support CMAKE_CUDA_ARCHITECTURES and CMAKE_HIP_ARCHITECTURES. Is there a similar native CMake option for clang-cuda? Should we be steering folks to use the native CMake variables instead of BLT variables?

Reply:
The clang cuda functionality certainly doesnt match up with the regular paths. BLT doesn't do anything special with the arch flags except in this case. Which I agree is not consistent.

Reply:
I'm debating not passing down the `CMAKE_CUDA_ARCHITECTURES` flag at all because I don't think it should be necessary for correctly setting up a CUDA target from a downstream library.  I think users tend to provide this variable via a host config.  In this case, even if I did include logic here for that, any variable provided in the host config would override it.